openeo/rest/datacube.py

```diff
@@ -1829,6 +1830,71 @@ def validate(self) -> List[dict]:
     def tiled_viewing_service(self, type: str, **kwargs) -> Service:
         return self._connection.create_service(self.flat_graph(), type=type, **kwargs)
 
+    def _get_spatial_extent_from_load_collection(self):
+        pg = self.flat_graph()
+        for node in pg:
+            if pg[node]["process_id"] == "load_collection":
+                if "spatial_extent" in pg[node]["arguments"] and all(
+                    cd
+                    for cd in pg[node]["arguments"]["spatial_extent"]
+                    for cd in ["east", "west", "south", "north"]
+                ):
+                    return pg[node]["arguments"]["spatial_extent"]
+        return None
+
+
+    def preview(
+        self,
+        center: Union[Iterable, None] = None,
+        zoom: Union[int, None] = None,
+    ) -> Tuple[Map, Service]:
+        """
+        Creates a service with the process graph and displays a map widget. Only supports XYZ.
+
+        :param center: (optional) Map center. Default is (0,0).
+        :param zoom: (optional) Zoom level of the map. Default is 1.
+
+        :return: ipyleaflet Map object and the displayed Service
```

Comment:
I'm wondering if we can't just a single object here

e.g. a service object that also contains the ipyleaflet map (which) is rendered through the `_repr_html_` API

Reply:
Could conflict with the Vue components, but could be combined somehow, I guess.

Reply:
I added just a simple wrapper if that's ok.